uploadctl/cleanup/upload_cleaner.py

```diff
@@ -1,131 +1,114 @@
-
-    def _check_if_user_used_recently(self, user):
-        used_ago = self.now - self._user_last_used_at(user)
-        sys.stdout.write(f"user used {used_ago.days} days ago, ")
-        if used_ago.days <= self.clean_older_than_days:
-            print("skipping.")
-            raise self.DontDelete()
-        else:
-            self.counts['old_users'] += 1
-
-    def _user_last_used_at(self, user):
-        """ Return latest date that a user account was used or a file modified in the upload area """
-        last_used = user['CreateDate']
-        for key in self.iam.list_access_keys(UserName=user['UserName'])['AccessKeyMetadata']:
-            if key['CreateDate'] > last_used:
-                last_used = key['CreateDate']
-            key_last_used = self.iam.get_access_key_last_used(AccessKeyId=key['AccessKeyId'])['AccessKeyLastUsed']
-            if 'LastUsedDate' in key_last_used and key_last_used['LastUsedDate'] > last_used:
-                last_used = key_last_used['LastUsedDate']
-        return last_used
-
-    def _check_special_case_upload_areas(self, area_uuid):
-        if re.match('aaaaaaaa-bbbb-cccc-dddd-.*', area_uuid):
-            print("special case.")
-            raise self.DontDelete
-
-    def _check_if_files_modified_recently(self, upload_area):
-        if self.ignore_file_age:
-            return
-        used_ago = self.now - self._files_last_modified_at(upload_area)
-        sys.stdout.write(f"files used {used_ago.days} days ago, ")
-        if used_ago.days <= self.clean_older_than_days:
-            print("skipping.")
-            raise self.DontDelete()
-
-    def _files_last_modified_at(self, upload_area):
-        last_file_modified_at = datetime.fromtimestamp(0, tz=timezone.utc)
-        files = upload_area.ls()['files']
-        sys.stdout.write(f"{len(files)} files, ")
-        sys.stdout.flush()
-        for file in files:
-            file_last_modified = dateutil.parser.parse(file['last_modified'])
-            if file_last_modified > last_file_modified_at:
-                last_file_modified_at = file_last_modified
-        return last_file_modified_at
-
-    def _iam_users(self):
-        marker = None
-        while True:
-            if marker:
-                resp = self.iam.list_users(Marker=marker)
+    def __init__(self, options):
+        self.options = options
+        self.s3 = boto3.resource('s3')
+        self.bucket = self.s3.Bucket(f"org-humancellatlas-upload-{os.environ['DEPLOYMENT_STAGE']}")
+        self.db_session_maker = DBSessionMaker()
+
+    def clean_files(self):
+        """
+        Delete entries from file,checksum,validation,notification DB tables for which this is no file in S3.
+        For files that are is S3, ensure the s3_etag is up to date in the DB file record.
+        """
+        global stats
+
+        session = self.db_session_maker.session()
+        t1 = time.time()
+        files_id_list = []
+        # TODO: sam remove filter after this run
```

Comment:
not sure what this run is referring to --possibly remove filter?

Reply:
Oh, "this run" is the cleanup/deletion of stuff in dev/int/staging/prod, and the setting of s3_etag in every file record.  After this has been completed, we will make s3_etag a `NOT NULL` column, rendering the `.filter(DbFile.s3_etag == None)` filter pointless.  I shouldn't really leave comments like this in, but on reflection as we haven't completed the cleanup, it is kinda useful to leave in for now.

Reply:
ok that makes sense. Just double checking